### PR TITLE
[Feature/fetch departments flow]: Fixed departments list, departments data model

### PR DESCRIPTION
**Pull request by stoyar**

### CODE REVIEW

File: src/api/departments/sendCorrections.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+) => Promise<ApiResultType<CorrectionsResponse>>;
+
+export const sendCorrections: SendCorrectionsRequest = (corrections) =>
+  new Promise((resolve) => {
+    const result: ApiResultType<CorrectionsResponse> = {
+      response: null,
+      error: null,
+    };
+
+    console.log(corrections);
```
Comment (andriibodryi) [inline src/api/departments/sendCorrections.ts:19]: please, remove log

File: src/screens/PriceError/elements/RequestStatusModal/RequestStatusModal.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        )}
+        {isLoading ? (
+          <Button
+            onPress={onCancelPress}
+            primary
+            small
+            text={CANCEL}
+            style={styles.button}
+          />
+        ) : (
```
Comment (andriibodryi) [inline src/screens/PriceError/elements/RequestStatusModal/RequestStatusModal.tsx:46]: may you combine this two elements into one isLoading?

**Review by andriibodryi — CHANGES_REQUESTED** [review]

File: src/api/departments/sendCorrections.ts (hunk 1 above)
Comment (stoyar) [reply src/api/departments/sendCorrections.ts:19]: This is temporary, because of ESLint Error for unused props

**Review by andriibodryi — APPROVED** [review]

File: src/screens/PriceError/elements/RequestStatusModal/RequestStatusModal.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  onClose,
+  isOpen,
+  isLoading,
+  requestStatus,
+}) => {
+  return (
+    <BottomSheetModal isOpen={isOpen} snapPoints={[0, 300]} onClose={onClose}>
+      <View style={styles.container}>
+        <Loader isLoading={isLoading} size={50} />
+        {isLoading || (
```
Comment (Mykhailo-Kaduk) [inline src/screens/PriceError/elements/RequestStatusModal/RequestStatusModal.tsx:33]: please move this piece of code into condition below

File: src/screens/PriceError/PriceError.tsx [hunk 4]
```diff
@@ -38,19 +40,62 @@ export const PriceError: React.FC = () => {
   }, []);
 
   const handleFinishPress = useCallback(() => {
+    sendCorrections(corrections);
+    setShowRequestStatus(true);
```
Comment (Mykhailo-Kaduk) [inline src/screens/PriceError/PriceError.tsx:44]: Maybe you should use the value from store instead of setting it from inside the component

Comment (stoyar) [reply src/screens/PriceError/PriceError.tsx:44]: This is not the isLoading value. I use this state for show/hide modal with request status and controls. isLoading don't fit for this purpose

File: src/screens/PriceError/elements/AddDepartmentModal/AddDepartmentModal.tsx [hunk 5]
```diff
@@ -47,17 +47,20 @@ export const AddDepartmentModal: React.FC<AddDepartmentProps> = ({
...
-        </Text>
-      </TouchableOpacity>
-    );
-  };
+  const renderDepartment = useCallback(
+    ({ item }) => {
+      const style =
+        department === item.id ? styles.selectedItem : styles.listItem;
+      return (
+        <TouchableOpacity onPress={() => setDepartment(item.id)} style={style}>
```
Comment (Mykhailo-Kaduk) [inline src/screens/PriceError/elements/AddDepartmentModal/AddDepartmentModal.tsx:55]: please make this callback memoized)

**Review by Mykhailo-Kaduk — APPROVED** [review]

**Pull request merged**